PIP Install issue for leveldb with python3.4.2

I am getting an error very similar to the one described here . https://groups.google.com/forum/#!topic/py-leveldb/VpVtc9hJatw . I am installing the package using pip. I also tried the solution provided in the forum and works fine for me. Can you please fix this so that pip3 install leveldb doesn't doesnt throw errors.

Here is what I get ::

    gcc -pthread -Wno-unused-result -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python3.4m -c leveldb/util/options.cc -o build/temp.linux-x86_64-3.4/leveldb/util/options.o -I./leveldb/include -I./leveldb -I./snappy -I. -fno-builtin-memcmp -O2 -fPIC -DNDEBUG -DSNAPPY -pthread -Wall -DOS_LINUX -DLEVELDB_PLATFORM_POSIX
    gcc -pthread -Wno-unused-result -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python3.4m -c leveldb/util/status.cc -o build/temp.linux-x86_64-3.4/leveldb/util/status.o -I./leveldb/include -I./leveldb -I./snappy -I. -fno-builtin-memcmp -O2 -fPIC -DNDEBUG -DSNAPPY -pthread -Wall -DOS_LINUX -DLEVELDB_PLATFORM_POSIX
    gcc -pthread -Wno-unused-result -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python3.4m -c leveldb/port/port_posix.cc -o build/temp.linux-x86_64-3.4/leveldb/port/port_posix.o -I./leveldb/include -I./leveldb -I./snappy -I. -fno-builtin-memcmp -O2 -fPIC -DNDEBUG -DSNAPPY -pthread -Wall -DOS_LINUX -DLEVELDB_PLATFORM_POSIX
    gcc -pthread -Wno-unused-result -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python3.4m -c leveldb_ext.cc -o build/temp.linux-x86_64-3.4/leveldb_ext.o -I./leveldb/include -I./leveldb -I./snappy -I. -fno-builtin-memcmp -O2 -fPIC -DNDEBUG -DSNAPPY -pthread -Wall -DOS_LINUX -DLEVELDB_PLATFORM_POSIX
    In file included from /usr/include/python3.4m/pyatomic.h:8:0,
                     from /usr/include/python3.4m/Python.h:53,
                     from leveldb_ext.h:8,
                     from leveldb_ext.cc:6:
    /usr/include/python3.4m/dynamic_annotations.h:472:3: error: template with C linkage
       template <class T>
       ^
    error: command 'gcc' failed with exit status 1

    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-lv3zxucm/leveldb/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-2x2_2bpt-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-lv3zxucm/leveldb/

--- a/leveldb_ext.h
+++ b/leveldb_ext.h
@@ -4,9 +4,11 @@
 #ifndef __LEVELDB__MODULE__H__
 #define __LEVELDB__MODULE__H__
 
-extern "C" {
+extern "C++" {
 #include <Python.h>
+}
 
+extern "C" {
 #include "structmember.h"
 
 #include <stdlib.h>

